OpenCL C 3.0 macros and cl_khr_3d_image_writes

LLVM's inclusion of OpenCL C 3.0 feature macros (see [d88c55](https://github.com/llvm/llvm-project/commit/d88c55ab95c9985e366b62569f2a50cf8ce5c469#)) broke parsing of the builtin headers when using OpenCL C 3.0 as the language standard.

Example errors:
```
In file included from <built-in>:375:
include/opencl-c.h:10571:37: error: use of type '__write_only image3d_t' requires cl_khr_3d_image_writes extension to be enabled
void __ovld write_imagef(write_only image3d_t image, int4 coord, float4 color);
                                    ^
```

Current workaround is to undef `cl_khr_3d_image_writes`. Longer term, figure what should actually be done about this.

Additionally, determine which feature macros should be defined for clspv. Current thoughts:
* __opencl_c_atomic_order_acq_rel
* __opencl_c_images (if images are supported)
* __opencl_c_3d_image_writes (if images are supported)
* __opencl_c_read_write_images (if images are supported)
* __opencl_c_subgroups (missing a couple features for full support)

--- a/lib/Compiler.cpp
+++ b/lib/Compiler.cpp
@@ -393,11 +393,6 @@ clang::TargetInfo *PrepareTargetInfo(CompilerInstance &instance) {
   // Disable CL3.0 feature macros for unsupported features
   if (instance.getLangOpts().LangStd == clang::LangStandard::lang_opencl30) {
 
-    // TODO: See #705, find a better solution for this.
-    // TODO(kpet): This is a Clang bug (the pragma isn't enabled when the macro
-    // is defined)
-    Opts["cl_khr_3d_image_writes"] = false;
-
     // The following features are never supported
     Opts["__opencl_c_pipes"] = false;
     Opts["__opencl_c_generic_address_space"] = false;

